### ISSUE TITLE
ipptransform crashes attempting to convert pdf to urf

### ISSUE THREAD

**Issue by tobeypasheilich** [istopwg/ippsample#226]
I'm attempting to use ipptransform to convert pdf documents to image/urf (or pwg) format (to send to a printer), but ipptransform is crashing on my system.  Full disclosure: I'm attempting this on a Windows box running Ubuntu 18.04 via WSL2 (Windows Subsystem for Linux).  I don't think that should matter, but want to be up front about it.

On 4/25/2021 I pulled down the ippsample source and installed the dependencies via the apt-get command noted in Build.md.  Then I ran ./configure, make, and then attempted to transform one of the sample pdf files to auf format via:
./ipptransform -f converted.urf -m image/urf ../examples/onepage-letter.pdf

Log of commands / output attached.  [Log.txt](https://github.com/istopwg/ippsample/files/6373234/Log.txt)

Ultimately I get a seg fault.  Am I doing something wrong or have I made a bad assumption about how this should work?  My ultimate goal is to get this working as a native Windows executable .  I'm able to build one, but it's crashing.  I tried debugging for quite a while, but eventually decided to confirm that this works as I'm expecting in a Linux environment, but I'm finding it behaves the same as my Windows compilation.

**Comment by michaelrsweet** [comment]
@tobeypasheilich What version of MuPDF do you have installed?

**Comment by tobeypasheilich** [comment]
Looks like 1.12.0_ds1-1 in my linux environment.  For my Windows work, I pulled down the current GitHub code (as of yesterday).

Linux info:
tobey@Tobey7530:/mnt/c/Projects/NeuMoDx/GitHub/Linux/ippsample/tools$ apt-show-versions libmupdf-dev
libmupdf-dev:amd64/bionic 1.12.0+ds1-1 uptodate
tobey@Tobey7530:/mnt/c/Projects/NeuMoDx/GitHub/Linux/ippsample/tools$ apt show libmupdf-dev
Package: libmupdf-dev
Version: 1.12.0+ds1-1
Priority: optional
Section: universe/libdevel
Source: mupdf
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Kan-Ru Chen (陳侃如) <koster@debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 48.2 MB
Homepage: http://mupdf.com/
Download-Size: 21.5 MB
APT-Manual-Installed: yes
APT-Sources: http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
Description: development files for the MuPDF viewer
 MuPDF is a lightweight PDF viewer and toolkit written in portable C.
 It also reads XPS, OpenXPS and ePub documents.
 .
 This package contains the static library and headers.

**Comment by michaelrsweet** [comment]
It is entirely possible you are running into a MuPDF bug - some versions have yielded segfaults for me before, and the API stability/instability has made keeping up with things very hard. I'll see if I can reproduce the problem, but no promises...

**Comment by tobeypasheilich** [comment]
@michaelrsweet I appreciate this very much!!  It seems like I must be doing something wrong, but I'm not seeing / understanding it.

There's clearly a lot going on in the mupdf library and I don't know my way around there.  One symptom / oddity I've noticed on both of the input PDF files I've tried is, I end up in fz_scan_convert_sharp (in mupdf/source/fitz/draw-edge.c, line 826 in current source) with a negative value for height (-37849 in one case).  Which means this loop will execute until h underflows and rolls all the way down to 0...which seems very wrong to me, but...

			h = height;
			while (h--)
			{
				if (eofill)
					even_odd_sharp(ctx, gel, y, clip, dst, color, fn, eop);
				else
					non_zero_winding_sharp(ctx, gel, y, clip, dst, color, fn, eop);
				y++;
			}

It seems reasonable to assume the error happened well before here as height is determined in part from the value of gel->edges[0].y.  In this case that y value is 40999 for me, which seems out of expected range.

This is as far as I've back tracked thus far, but I really don't have much idea what I'm looking at. :)

I plan to continue digging, but I'm just not familiar with what this is doing / how it works / etc.

It would be great to know if you're able to transform the sample file in your environment.  I'm doing most of my work in my Windows environment as that's what I'm more familiar with.
